src/plugins/score-plugin-deviceexplorer/Explorer/Explorer/Widgets/DeviceEditDialog.cpp

```diff
@@ -58,65 +59,76 @@ DeviceEditDialog::DeviceEditDialog(
   };
 
   setWindowTitle(tr("Add device"));
-  auto gridLayout = new QGridLayout{};
-  setLayout(gridLayout);
+  auto base_layout = new QVBoxLayout{this};
+  setLayout(base_layout);
   setModal(true);
 
-  m_invalidLabel = new QLabel{
-      tr("Cannot add device.\n Try changing the name to make it unique, \nor "
-         "check that the ports aren't already used")};
-  m_invalidLabel->setAlignment(Qt::AlignRight);
-  m_invalidLabel->setTextFormat(Qt::PlainText);
-  m_buttonBox = new QDialogButtonBox(Qt::Horizontal, this);
-  m_okButton = m_buttonBox->addButton(tr("Add"), QDialogButtonBox::AcceptRole);
-  m_buttonBox->addButton(QDialogButtonBox::Cancel);
-
-  connect(m_buttonBox, &QDialogButtonBox::accepted, this, &QDialog::accept);
-  connect(m_buttonBox, &QDialogButtonBox::rejected, this, &QDialog::reject);
+  auto splitter = new QSplitter{this};
+  base_layout->addWidget(splitter);
 
+  auto column1 = new QWidget;
+  auto column1_layout = new QVBoxLayout{column1};
+  auto protocolsLabel = new QLabel{tr("Protocols"), this};
+  setHeaderTextFormat(protocolsLabel);
+  column1_layout->addWidget(protocolsLabel);
+  protocolsLabel->setAlignment(Qt::AlignTop);
+  protocolsLabel->setAlignment(Qt::AlignHCenter);
   m_protocols = new QTreeWidget{this};
   m_protocols->header()->hide();
   m_protocols->setSelectionMode(QAbstractItemView::SingleSelection);
+  column1_layout->addWidget(m_protocols);
+  column1->setLayout(column1_layout);
+  splitter->addWidget(column1);
+
   if(m_mode == Mode::Editing)
   {
-    m_protocols->setVisible(false);
-  }
-  else
-  {
-    m_protocols->setFixedWidth(150);
-
-    {
-      m_protocolsLabel = new QLabel{tr("Protocols"), this};
-      setHeaderTextFormat(m_protocolsLabel);
-      gridLayout->addWidget(m_protocolsLabel, 0, 0, Qt::AlignHCenter);
-    }
-    gridLayout->addWidget(m_protocols, 1, 0);
+    column1->setVisible(false);
   }
 
+  // Column 2: Devices
+  auto column2 = new QWidget;
+  auto column2_layout = new QVBoxLayout{column2};
+  auto devicesLabel = new QLabel{tr("Devices"), this};
+  setHeaderTextFormat(devicesLabel);
+  column2_layout->addWidget(devicesLabel);
+  devicesLabel->setAlignment(Qt::AlignTop);
+  devicesLabel->setAlignment(Qt::AlignHCenter);
   m_devices = new QTreeWidget{this};
   m_devices->header()->hide();
   m_devices->setSelectionMode(QAbstractItemView::SingleSelection);
-  // m_devices->setFixedWidth(140);
-  gridLayout->addWidget(m_devices, 1, 1);
-  {
-    m_devicesLabel = new QLabel{tr("Devices"), this};
-    setHeaderTextFormat(m_devicesLabel);
-    gridLayout->addWidget(m_devicesLabel, 0, 1, Qt::AlignHCenter);
-  }
-
-  {
-    m_protocolNameLabel = new QLabel{tr("Settings"), this};
-    setHeaderTextFormat(m_protocolNameLabel);
-    gridLayout->addWidget(
-        m_protocolNameLabel, 0, 2, -1, -1, Qt::AlignLeft | Qt::AlignTop);
-  }
+  qDebug() << "m_devices initialized : " << m_devices;
```

Comment:
For a PR this shouldn't have qDebug

Reply:
I wasn't readdyyyy you pushed on my behalf :[  this is a draft

Reply:
Hu ? I didn't push anything